### DISCUSSION TITLE
Feature Request: FireTickChange event

### DISCUSSION THREAD

**Discussion by TetraTheta** [PaperMC/Paper#10522] [Ideas]
### Is your feature request related to a problem?

In Minecraft, if a player has 'Fire Resistance' effect on him and get into lava(or fire), he gets on fire. But since 'Fire Resistance' is blocking him to get damaged, `EntityDamageEvent` won't fire.

In this state, there is no way to detect player is on fire or not via event. Running task every tick would be terrible idea, but I think this is the only way to detect 'Player is on fire' currently.

### Describe the solution you'd like.

So, I think it would be great if there is some kind of `FireTickChangeEvent` exists. It would be fired if a player's fire tick is changed, even if the player don't take any damage from the fire/lava.

### Describe alternatives you've considered.

The main reason I wrote this Feature Request is get a way to detect player on fire, even if he has 'Fire Resistance' effect on him properly, without using task running every tick(s). So It doesn't matter what the event's name would be. I am terrible with naming stuff...

In addition, if the `FireTickChangeEvent` can be applied to mobs, that would be more generic I guess... (For Entity, not only for Player)

### Agreements

- [X] I have searched for and ensured there isn't already an open issue regarding this.
- [X] I have ensured the feature I'm requesting isn't already in the latest supported Paper build.

**Comment by aurorasmiles** [comment]
What you want might be possible with #5242

**Comment by TetraTheta** [comment]
> 
> 
> What you want might be possible with #5242

If then, it should be extended for both `onFire`(FireTickChangeEvent) and `onExtinguish`(EntityExtinguishEvent) to detect 'Does it is on fire?' I guess. Did you mean this?

**Comment by LysergikProductions** [comment]
I second this need for a `FireTickChange` or `FireTickUpdateEvent` kind of event for all entity objects.
On top of that, there also appears to be no way to check if a given `Block` object is currently burning or not.

`FireTickUpdateEvent` could have primitives like `int fireAge`, `boolean queuedForExtinguish`, `boolean isNewFire` (true if is youngest possible fire age on trigger), etc, as well as methods like `Block getBlock()` (resulting in null if entity), `Entity getEntity()` (resulting in null if block), `boolean setFireAge(int fire_age)`, etc
